Week 5/Solution2.java

```diff
@@ -0,0 +1,109 @@
+import java.io.*;
+import java.util.*;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+//This class is the implementation of the alternative method with O(1) space usage.
+//Implies the modification of the list as suggested by Connie and has the fast pointer method to find the middle node.
+public class Solution2 {
+	//Node structure considered for this problem
+	class Node{
+		char value;
+		Node next;
+		Node(char value, Node next){
+			this.value = value;
+			this.next = next;
+		}
+	}
+    public boolean isPalindrome(Node head) {
+    	//input validation
+    	if(head == null) throw new NullPointerException();
+    	//getting the middle element and length of the list to know if it is even or odd
+        int ln = getLength(head);
+    	Node middle = getMiddle(head);
+        //if odd we ignore the middle element
+    	if(ln%2!=0) middle = middle.next;
```

Comment:
Food for thought - you can actually get away with not calculating the length at the beginning. You have almost all of the pieces needed for that solution here already :) Hint - how would using pointers moving at different speeds allow you to determine the middle element _and_ also give you the information necessary to tell if the list is odd?

Reply:
Thanks for the retro! I'll correct it as soon as possible :)